js/index.js

```diff
@@ -9,15 +22,26 @@ function updateSubtotal(product) {
 function calculateAll() {
   // code in the following two lines is added just for testing purposes.
   // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
+  // const singleProduct = document.querySelector('.product');
+  const productElements = document.getElementsByClassName('product');
+  let totalValue = 0;
+  for (productElement of productElements) {
+    totalValue += updateSubtotal(productElement)
+  }
+  //updateSubtotal(singleProduct);
   // end of test
 
-  // ITERATION 2
-  //... your code goes here
 
-  // ITERATION 3
-  //... your code goes here
+
+  // call the function updateSubtotal with every tr.product DOM node in the table#cart
+  // get cart item elements getElementsByClassName
+  //for product of products call updateSubtotal(product)
+
+  // reuse total value of it 2
+  // update dom element
+  // calculate totalprice by summing all of subtotal returned by updateSubtotal()
+  const totalValueSpan = document.querySelector('#total-value span').innerHTML = totalValue;
+
 }
```

Comment:
Great stuff Sabela!
Please check whether you still need the comments and console.log's.